src/domain/checkForAndPlaceOrder.ts

```diff
@@ -533,152 +533,155 @@ async function _pollLegacy(
-    return _handleGetTradableOrderWithSignatureCall(error, owner, orderRef);
   }
 }
 
-function _handleGetTradableOrderWithSignatureCall(
-  error: any,
+function _handleOnChainCustomError(
   owner: string,
-  orderRef: string
+  orderRef: string,
+  context: ChainContext,
+  to: string,
+  data: string,
+  returnData: string
 ): PollResultErrors {
-  const logPrefix = `checkForAndPlaceOrder:_handleGetTradableOrderCall:${orderRef}`;
+  const logPrefix = `checkForAndPlaceOrder:_handleOnChainCustomError:${orderRef}`;
   const log = getLogger(logPrefix);
+  const { chainId } = context;
 
-  // If the error is a LowLevelError, we extract the selector, and any parameters.
-  if (error instanceof LowLevelError) {
-    try {
-      // The below will throw if:
-      // - the error is not a custom error (ie. the selector is not in the map)
-      // - the error is a custom error, but the parameters are not as expected
-      const { selector, message, blockNumberOrEpoch } = customErrorDecode(
-        error.data
-      );
-      switch (selector) {
-        case "SINGLE_ORDER_NOT_AUTHED":
-        case "PROOF_NOT_AUTHED":
-          // If there's no authorization we delete the order
-          // - One reason could be, because the user CANCELLED the order
-          // - for now it doesn't support more advanced cases where the order is auth during a pre-interaction
-          log.info(
-            `${selector}: Order on safe ${owner} not authed. Deleting order...`
-          );
-          return {
-            result: PollResultCode.DONT_TRY_AGAIN,
-            reason: `${selector}: The owner has not authorized the order`,
-          };
-        case "INTERFACE_NOT_SUPPORTED":
-          log.info(
-            `${selector}: Order on safe ${owner} attempted to use a handler that is not supported. Deleting order...`
-          );
-          return {
-            result: PollResultCode.DONT_TRY_AGAIN,
-            reason: `${selector}: The handler is not supported`,
-          };
-        case "INVALID_FALLBACK_HANDLER":
-          log.info(
-            `${selector}: Order for safe ${owner} where the Safe does not have ExtensibleFallbackHandler set. Deleting order...`
-          );
-          return {
-            result: PollResultCode.DONT_TRY_AGAIN,
-            reason: `${selector}: The safe does not have ExtensibleFallbackHandler set`,
-          };
-        case "SWAP_GUARD_RESTRICTED":
-          log.info(
-            `${selector}: Order for safe ${owner} where the Safe has swap guard enabled. Deleting order...`
+  try {
+    // The below will throw if:
+    // - the error is not a custom error (ie. the selector is not in the map)
+    // - the error is a custom error, but the parameters are not as expected
+    const { selector, message, blockNumberOrEpoch } =
+      customErrorDecode(returnData);
+    switch (selector) {
+      case "SINGLE_ORDER_NOT_AUTHED":
+      case "PROOF_NOT_AUTHED":
```

Comment:
We have this literal in 3 points in the app. Should we make an enum or sth to make sure typescript prevents us from making mistakes and we can keep the code dry?

Reply:
Enums have been implemented in typescript, and custom error handler has been moved to `contract.ts` within `utils`.